pyth_node/README.md

```diff
@@ -0,0 +1,104 @@
+Pyth Node
+================================================================================
+
+Pyth Node is a highly available and versatile software that defines the Pyth
+P2P network, Pyth API, and connects with the Pyth Geyser node software to
+listen for PythNet events. It offers both REST and Websocket APIs for ease of
+integration and interaction with the Pyth network stack. The Node can be run by
+anyone who wishes to participate / observe the PythNet network operations.
+
+You can run this software if you wish to offer a PythNet API, or simply observe
+prices from the Pyth network for other purposes.
+
+Overview
+--------------------------------------------------------------------------------
+
+Pyth was initially designed with a simple "Price Service" with a REST API that
+could be used to request prices to submit to other networks. However, as the
+project evolves, its architecture can be rethought and improved to provide a
+more general node software that incorporates the Price Service as one of its
+components.
+
+This crate now provides this software.
+
+If you were previously running the Price Service it can be replaced by running
+this new node software. The new the new architecture for this service can be
+understood visually as so:
+
+---
+
+![image](https://user-images.githubusercontent.com/158967/225939587-f19cfe77-0393-4798-ad72-0022420d3e51.png)
+
+---
+
+This is more in line with other well-known projects in the blockchain and P2P
+space, such as Solana, Bitcoin, and Cosmos chains, where users who run the node
+can also act as observers with specific features disabled. In our case, the
+node software can be run with or without a PythNet validator running.
+
+Pyth Node has several responsibilities:
+
+- Offering a REST and Websocket API to interact with the Pyth network.
+- Connecting to the Wormhole P2P network and listening for Pyth-related VAAs/Messages.
+- Connecting to the Pyth P2P network and sharing Pyth-specific Account Updates.
+- Connecting to PythNet itself via our Geyser plugin to allow for a fully
+  decentralised PythNet.
+
+The node is designed to fetch updates from either Pyth or Geyser networks for
+high availability.
+
+Architecture
+--------------------------------------------------------------------------------
+
+The Pyth Node software consists of the following components:
+
+- **Wormhole P2P Connector**: Connects to the Wormhole P2P network and listens for VAAs.
+- **Pyth P2P Connector**: Connects to the Pyth P2P network and listens for Account Updates.
+- **Geyser Connector**: Connects to the Geyser node software and listens for Account Updates.
+- **REST API**: Provides an interface for external applications to interact with PythNet.
+- **Websocket API**: Offers real-time data streaming for interacting with PythNet.
```

Comment:
you may want to say something like "the Pyth Node is designed to be run in two configurations: standalone, or alongside a PythNet validator". No one is going to use both the P2P and the Geyser connections at the same time, so probably makes sense to differentiate those as separate use cases.